client/src/main/java/crawlercommons/urlfrontier/client/GetURLStatus.java

```diff
@@ -0,0 +1,101 @@
+package crawlercommons.urlfrontier.client;
+
+import crawlercommons.urlfrontier.URLFrontierGrpc;
+import crawlercommons.urlfrontier.URLFrontierGrpc.URLFrontierBlockingStub;
+import crawlercommons.urlfrontier.Urlfrontier.URLItem;
+import crawlercommons.urlfrontier.Urlfrontier.URLStatusRequest;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.StatusRuntimeException;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import picocli.CommandLine.ArgGroup;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.ParentCommand;
+
+@Command(name = "GetURLStatus", description = "Get the status of an URL")
+public class GetURLStatus implements Runnable {
+
+    @ParentCommand private Client parent;
+
+    @ArgGroup(exclusive = false, multiplicity = "1")
+    Dependent group;
+
+    static class Dependent {
+        @Option(
+                names = {"-c", "--crawlID"},
+                required = true,
+                paramLabel = "STRING",
+                description = "crawl of the url to be checked")
+        private String crawl;
+
+        @Option(
+                names = {"-k", "--key"},
```

Comment:
see comment about key being generated by default on the server side. Should be optional?

Reply:
ok